controllers/disruption_controller.go

```diff
@@ -600,32 +600,27 @@ func (r *DisruptionReconciler) selectTargets(instance *chaosv1beta1.Disruption)
 		}
 	}
 
-	// prune ignored targets from all targets
-	for _, target := range matchingTargets {
-		if !contains(instance.Status.IgnoredTargets, target) {
-			eligibleTargets = append(eligibleTargets, target)
-		}
-	}
-
 	// return an error if the selector returned no targets
-	if len(eligibleTargets) == 0 {
+	if len(matchingTargets) == 0 {
 		r.log.Info("the given label selector did not return any targets, skipping")
 		r.Recorder.Event(instance, "Warning", "NoTarget", "The given label selector did not return any targets. Please ensure that both the selector and the count are correct (should be either a percentage or an integer greater than 0).")
 
 		return nil
 	}
 
+	// prune ignored targets from all targets
+	for _, target := range matchingTargets {
+		if !contains(instance.Status.IgnoredTargets, target) {
+			eligibleTargets = append(eligibleTargets, target)
+		}
+	}
+
 	// instance.Spec.Count is a string that either represents a percentage or a value, we do the translation here
-	targetsCount, err := getScaledValueFromIntOrPercent(instance.Spec.Count, len(eligibleTargets), true)
+	targetsCount, err := getScaledValueFromIntOrPercent(instance.Spec.Count, len(matchingTargets), true)
```

Comment:
Imagine the case where count = 100%, matching = 5, ignored = 3, and eligible = 2. In that situation, we're going to get `2` as the targetsCount, where we will immediately subtract 3, get a final count of -1 and end the disruption early. We should expect when count = 100% that we will target len(matching), which is now the behavior

Reply:
It is perhaps easier to explain the issue with the current code by saying that when we calculate a scaled percentage count, we are subtracting ignored, calculating the percentage from the reduced total, then subtracting ignored again. This is causing us to under-disrupt